src/services/cronJobs/checkProjectVerificationStatus.ts

```diff
@@ -0,0 +1,193 @@
+
+const maxDaysForSendingUpdateWarning = moment()
+  .subtract(projectUpdatesWarningDays, 'days')
+  .endOf('day')
+  .toDate();
+
+const maxDaysForSendingUpdateLastWarning = moment()
+  .subtract(projectUpdatesLastWarningDays, 'days')
+  .endOf('day')
+  .toDate();
+
+const maxDaysForRevokingBadge = moment()
+  .subtract(projectUpdatesRevokeVerificationDays, 'days')
+  .endOf('day')
+  .toDate();
+
+export const runCheckProjectVerificationStatus = () => {
+  logger.debug('runCheckProjectVerificationStatus() has been called');
+  schedule(cronJobTime, async () => {
+    await checkProjectVerificationStatus();
+  });
+};
+
+export const checkProjectVerificationStatus = async () => {
+  // all projects with last update created at 30+ days
+  const projects = await projectsWithoutUpdateAfterTimeFrame(
+    maxDaysForSendingUpdateReminder,
+  );
+
+  for (const project of projects) {
+    await remindUpdatesOrRevokeVerification(project);
+  }
+};
+
+const remindUpdatesOrRevokeVerification = async (project: Project) => {
+  // Projects up for revoking when 30 days are done after feature release
+  const lastChanceRevokeDate = new Date().setDate(
+    maxDaysForSendingUpdateLastWarning.getDate() -
+      projectUpdatesExpiredRevokeAdditionalDays,
+  );
+  if (
+    project.projectUpdate.createdAt <= lastChanceRevokeDate &&
+    project.verificationStatus === RevokeSteps.UpForRevoking
+  ) {
+    project.verificationStatus = RevokeSteps.Revoked;
+    project.verified = false;
+  }
+
+  // Projects that already expired verification are given a last chance
+  // for this feature
+  if (
+    project.projectUpdate.createdAt <= maxDaysForSendingUpdateLastWarning &&
+    project.verificationStatus === null
+  ) {
+    project.verificationStatus = RevokeSteps.UpForRevoking;
+  }
+
+  // Projects that had the last chance and failed to add an update are revoked
+  if (
+    project.projectUpdate.createdAt <= maxDaysForRevokingBadge &&
+    project.verificationStatus === RevokeSteps.LastChance
+  ) {
+    project.verificationStatus = RevokeSteps.Revoked;
+    project.verified = false;
+  }
+
+  // projects that were warned are not sent a last chance warning
+  if (
+    project.projectUpdate.createdAt <= maxDaysForSendingUpdateLastWarning &&
+    project.projectUpdate.createdAt > maxDaysForRevokingBadge &&
+    project.verificationStatus === RevokeSteps.Warning
+  ) {
+    project.verificationStatus = RevokeSteps.LastChance;
+  }
+
+  // After reminder at 60/75 days
+  if (
+    project.projectUpdate.createdAt <= maxDaysForSendingUpdateWarning &&
+    project.projectUpdate.createdAt > maxDaysForSendingUpdateLastWarning
+  ) {
+    project.verificationStatus = RevokeSteps.Warning;
+  }
+
+  // First email for reminding to add an update
+  if (
+    project.projectUpdate.createdAt <= maxDaysForSendingUpdateReminder &&
+    project.projectUpdate.createdAt > maxDaysForSendingUpdateWarning
+  ) {
+    project.verificationStatus = RevokeSteps.Reminder;
+  }
+
```

Comment:
Technically is not posible, I tested manually the conditions and thats why they are ordered that way and the status is the limiting factor so it doesn't get into two. And the ifs are by time range.

But I think else if might be safer, just in case.